dotlottie-rs/src/dotlottie_player.rs

```diff
@@ -1141,6 +1184,598 @@ impl DotLottiePlayer {
     pub fn active_theme_id(&self) -> String {
         self.runtime.read().unwrap().active_theme_id().to_string()
     }
+
+    pub fn get_state_machine(&self, state_machine_id: &str) -> Option<String> {
+        self.runtime
+            .read()
+            .unwrap()
+            .get_state_machine(state_machine_id)
+    }
+}
+
+pub struct DotLottiePlayer {
+    player: Rc<RwLock<DotLottiePlayerContainer>>,
+    state_machine: Rc<RwLock<Option<StateMachine>>>,
+}
+
+impl DotLottiePlayer {
+    pub fn new(config: Config) -> Self {
+        DotLottiePlayer {
+            player: Rc::new(RwLock::new(DotLottiePlayerContainer::new(config))),
+            state_machine: Rc::new(RwLock::new(None)),
+        }
+    }
+
+    pub fn load_animation_data(&self, animation_data: &str, width: u32, height: u32) -> bool {
+        let is_ok = self
+            .player
+            .write()
+            .is_ok_and(|runtime| runtime.load_animation_data(animation_data, width, height));
+        is_ok
+    }
+
+    pub fn get_state_machine(&self) -> Rc<RwLock<Option<StateMachine>>> {
+        self.state_machine.clone()
+    }
+
+    pub fn start_state_machine(&self) -> bool {
+        if self.state_machine.read().unwrap().is_none() {
+            return false;
+        }
+
+        match self.state_machine.try_write() {
+            Ok(mut state_machine) => {
+                if let Some(sm) = state_machine.as_mut() {
+                    sm.start();
```

Comment:
is there a possibility that the sm.start won't start ? i feel that the sm.start should return bool to help the DotLottiePlayer user to if it's actually started or not, now we're assuming that if we can't acquire a write lock, then we can't start the state machine, which might not be the case in some situations

Reply:
also if there is no state machine, shouldn't it return false

Reply:
@samuelOsborne Returning enums as a result can work, but it still doesn't give the exact cause. That's an issue I'm still facing with ThorVG. Receiving an "Unsuccessful" error or an "Unsupported" error sometimes isn't useful.